pkg/cable/wireguard/driver.go

```diff
@@ -370,6 +402,27 @@ func discoverInternalIP(cidrs []net.IPNet) (string, error) {
 	return "", nil
 }
 
+// look at local subnets, cluster cidrs, and vxlan subnets
+func (w *wireguard) discoverInternalIP(localSubnets []string, localCluster types.SubmarinerCluster) (string, error) {
+	cidrs := parseSubnets(localSubnets)
+	ip, err := matchLocalIP(cidrs)
+	if ip != "" {
+		return ip, nil
+	}
+	if err != nil {
+		return "", err
+	}
+
+	// no match, fallback to cluster cidr
```

Comment:
Can you please explain what we are trying to do here?

Reply:
See PR commit comment.

Reply:
Ack, as I mentioned above, I have a feeling that you can all-together remove the search via localsubnets and simply search for an interface that has IPAddress from clusterCIDR.